### ISSUE TITLE
Ubuntu Failed Build

### ISSUE THREAD

**Issue by arshlearn** [master-of-zen/Av1an#898]
Hello All, i was compiling av1an on ubuntu with cargo, but i am running into a ffmpeg-the-third error, my guess is that it is an error with a dependency of av1an, not in av1an itself. I had also tried building from master, but i observd the same error. Its also showing libavcodec not found, whereas when i was compiling bestsource vapoursynth plugin earlier, it was compiling with 0 issues. Here is the log for future purposes

```
Updating crates.io index
 Downloading crates ...
  Downloaded atty v0.2.14
  Downloaded vapoursynth-sys v0.4.0
   Compiling proc-macro2 v1.0.86
   Compiling unicode-ident v1.0.12
   Compiling libc v0.2.155
   Compiling autocfg v1.3.0
   Compiling crossbeam-utils v0.8.20
   Compiling serde v1.0.205
   Compiling memchr v2.7.4
   Compiling thiserror v1.0.63
   Compiling pkg-config v0.3.30
   Compiling rustversion v1.0.17
   Compiling rayon-core v1.12.1
   Compiling anyhow v1.0.86
   Compiling regex-syntax v0.8.4
   Compiling either v1.13.0
   Compiling cfg-if v1.0.0
   Compiling version_check v0.9.5
   Compiling glob v0.3.1
   Compiling bitflags v2.6.0
   Compiling fnv v1.0.7
   Compiling strsim v0.11.1
   Compiling ident_case v1.0.1
   Compiling minimal-lexical v0.2.1
   Compiling bindgen v0.69.4
   Compiling vcpkg v0.2.15
   Compiling syn v1.0.109
   Compiling lazycell v1.3.0
   Compiling shlex v1.3.0
   Compiling tinyvec_macros v0.1.1
   Compiling rustc-hash v1.1.0
   Compiling lazy_static v1.5.0
   Compiling semver v1.0.23
   Compiling serde_json v1.0.122
   Compiling log v0.4.22
   Compiling camino v1.1.7
   Compiling itoa v1.0.11
   Compiling unicode-bidi v0.3.15
   Compiling percent-encoding v2.3.1
   Compiling powerfmt v0.2.0
   Compiling ryu v1.0.18
   Compiling aligned-vec v0.5.0
   Compiling byteorder v1.5.0
   Compiling parking_lot_core v0.9.10
   Compiling vapoursynth-sys v0.4.0
   Compiling paste v1.0.15
   Compiling num_threads v0.1.7
   Compiling built v0.7.4
   Compiling heck v0.5.0
   Compiling bytes v1.7.1
   Compiling time-core v0.1.2
   Compiling num-conv v0.1.0
   Compiling utf8parse v0.2.2
   Compiling smallvec v1.13.2
   Compiling scopeguard v1.2.0
   Compiling once_cell v1.19.0
   Compiling unicode-width v0.1.13
   Compiling rustix v0.38.34
   Compiling arrayvec v0.7.4
   Compiling byte-slice-cast v1.2.2
   Compiling ffmpeg-the-third v2.0.1+ffmpeg-7.0
   Compiling portable-atomic v1.7.0
   Compiling linux-raw-sys v0.4.14
   Compiling bitflags v1.3.2
   Compiling plotters-backend v0.3.6
   Compiling is_terminal_polyfill v1.70.1
   Compiling new_debug_unreachable v1.0.6
   Compiling anstyle-query v1.1.1
   Compiling anstyle v1.0.8
   Compiling bitstream-io v2.5.0
   Compiling colorchoice v1.0.2
   Compiling noop_proc_macro v0.3.0
   Compiling home v0.5.9
   Compiling hashbrown v0.14.5
   Compiling clap_lex v0.7.2
   Compiling stfu8 v0.2.7
   Compiling iana-time-zone v0.1.60
   Compiling number_prefix v0.4.0
   Compiling pin-project-lite v0.2.14
   Compiling std_prelude v0.2.12
   Compiling smawk v0.3.2
   Compiling unicode-linebreak v0.1.5
   Compiling y4m v0.8.0
   Compiling nu-ansi-term v0.50.1
   Compiling splines v4.3.1
   Compiling simdutf8 v0.1.4
   Compiling ansi_term v0.12.1
   Compiling libloading v0.8.5
   Compiling itertools v0.12.1
   Compiling form_urlencoded v1.2.1
   Compiling tinyvec v1.8.0
   Compiling anstyle-parse v0.2.5
   Compiling plotters-svg v0.3.6
   Compiling num-traits v0.2.19
   Compiling deranged v0.3.11
   Compiling lock_api v0.4.12
   Compiling itertools v0.13.0
   Compiling proc-macro-error-attr v1.0.4
   Compiling proc-macro-error v1.0.4
   Compiling clang-sys v1.8.1
   Compiling aho-corasick v1.1.3
   Compiling nom v7.1.3
   Compiling anstream v0.6.15
   Compiling textwrap v0.16.1
   Compiling clap_builder v4.5.15
   Compiling crossbeam-epoch v0.9.18
   Compiling quote v1.0.36
   Compiling vergen-lib v0.1.3
   Compiling vergen v9.0.0
   Compiling vergen-git2 v1.0.0
   Compiling jobserver v0.1.32
   Compiling time v0.3.36
   Compiling syn v2.0.72
   Compiling getrandom v0.2.15
   Compiling console v0.15.8
   Compiling num_cpus v1.16.0
   Compiling socket2 v0.5.7
   Compiling signal-hook-registry v1.4.2
   Compiling mio v1.0.1
   Compiling errno v0.3.9
   Compiling atty v0.2.14
   Compiling crossbeam-deque v0.8.5
   Compiling simd_helpers v0.1.0
   Compiling unicode-normalization v0.1.23
   Compiling rand_core v0.6.4
   Compiling crossbeam-channel v0.5.13
   Compiling cc v1.1.8
   Compiling affinity v0.1.2
   Compiling parking_lot v0.12.3
   Compiling num-integer v0.1.46
   Compiling v_frame v0.3.8
   Compiling chrono v0.4.38
   Compiling plotters v0.3.6
   Compiling tokio v1.39.2
   Compiling indicatif v0.17.8
   Compiling num-bigint v0.4.6
   Compiling idna v0.5.0
   Compiling rayon v1.10.0
   Compiling regex-automata v0.4.7
   Compiling url v2.5.2
   Compiling cexpr v0.6.0
   Compiling libz-sys v1.1.18
   Compiling libgit2-sys v0.17.0+1.8.1
   Compiling which v6.0.2
   Compiling getset v0.1.2
   Compiling num-rational v0.4.2
   Compiling av1-grain v0.2.3
   Compiling nasm-rs v0.2.5
   Compiling rav1e v0.7.1
   Compiling regex v1.10.6
   Compiling git2 v0.19.0
   Compiling darling_core v0.20.10
   Compiling maybe-rayon v0.1.1
   Compiling sysinfo v0.31.2
   Compiling serde_derive v1.0.205
   Compiling thiserror-impl v1.0.63
   Compiling num-derive v0.4.2
   Compiling zerocopy-derive v0.7.35
   Compiling profiling-procmacros v1.0.15
   Compiling arg_enum_proc_macro v0.3.4
   Compiling strum_macros v0.26.4
   Compiling clap_derive v4.5.13
   Compiling profiling v1.0.15
   Compiling zerocopy v0.7.35
   Compiling darling_macro v0.20.10
   Compiling av-data v0.4.2
   Compiling av-bitstream v0.2.0
   Compiling vapoursynth v0.4.0
   Compiling flexi_logger v0.28.5
   Compiling ppv-lite86 v0.2.20
   Compiling darling v0.20.10
   Compiling av-format v0.7.0
   Compiling derive_builder_core v0.20.0
   Compiling av-ivf v0.5.0
   Compiling clap v4.5.15
   Compiling rand_chacha v0.3.1
   Compiling rand v0.8.5
   Compiling strum v0.26.3
   Compiling derive_builder_macro v0.20.0
   Compiling derive_builder v0.20.0
   Compiling ffmpeg-sys-the-third v2.0.0+ffmpeg-7.0
   Compiling cargo-platform v0.1.8
   Compiling rustc_version v0.4.0
   Compiling dashmap v6.0.1
   Compiling path_abs v0.5.1
   Compiling cargo_metadata v0.18.1
   Compiling av1an v0.4.1 (/content/Av1an-0.4.3/av1an)
error[E0425]: cannot find value `SWS_FAST_BILINEAR` in this scope
 --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:6:38
  |
6 |         const FAST_BILINEAR        = SWS_FAST_BILINEAR;
  |                                      ^^^^^^^^^^^^^^^^^ not found in this scope
  |
help: consider importing one of these unit variants
  |
1 + use crate::software::scaling::SwsFlags::SWS_FAST_BILINEAR;
  |
1 + use ffmpeg_sys_the_third::SwsFlags::SWS_FAST_BILINEAR;
  |

error[E0425]: cannot find value `SWS_BILINEAR` in this scope
 --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:7:38
  |
7 |         const BILINEAR             = SWS_BILINEAR;
  |                                      ^^^^^^^^^^^^ not found in this scope
  |
help: consider importing one of these unit variants
  |
1 + use crate::software::scaling::SwsFlags::SWS_BILINEAR;
  |
1 + use ffmpeg_sys_the_third::SwsFlags::SWS_BILINEAR;
  |

warning: av1an@0.4.1: could not find repository at '/content/Av1an-0.4.3/av1an'; class=Repository (6); code=NotFound (-3)
warning: av1an@0.4.1: VERGEN_GIT_COMMIT_DATE set to default
warning: av1an@0.4.1: VERGEN_GIT_SHA set to default
error[E0425]: cannot find value `SWS_BICUBIC` in this scope
 --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:8:38
  |
8 |         const BICUBIC              = SWS_BICUBIC;
  |                                      ^^^^^^^^^^^ not found in this scope
  |
help: consider importing one of these unit variants
  |
1 + use crate::software::scaling::SwsFlags::SWS_BICUBIC;
  |
1 + use ffmpeg_sys_the_third::SwsFlags::SWS_BICUBIC;
  |

error[E0425]: cannot find value `SWS_X` in this scope
 --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:9:38
  |
9 |         const X                    = SWS_X;
  |                                      ^^^^^ not found in this scope
  |
help: consider importing one of these unit variants
  |
1 + use crate::software::scaling::SwsFlags::SWS_X;
  |
1 + use ffmpeg_sys_the_third::SwsFlags::SWS_X;
  |

error[E0425]: cannot find value `SWS_POINT` in this scope
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:10:38
   |
10 |         const POINT                = SWS_POINT;
   |                                      ^^^^^^^^^ not found in this scope
   |
help: consider importing one of these unit variants
   |
1  + use crate::software::scaling::SwsFlags::SWS_POINT;
   |
1  + use ffmpeg_sys_the_third::SwsFlags::SWS_POINT;
   |

error[E0425]: cannot find value `SWS_AREA` in this scope
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:11:38
   |
11 |         const AREA                 = SWS_AREA;
   |                                      ^^^^^^^^ not found in this scope
   |
help: consider importing one of these unit variants
   |
1  + use crate::software::scaling::SwsFlags::SWS_AREA;
   |
1  + use ffmpeg_sys_the_third::SwsFlags::SWS_AREA;
   |

error[E0425]: cannot find value `SWS_BICUBLIN` in this scope
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:12:38
   |
12 |         const BICUBLIN             = SWS_BICUBLIN;
   |                                      ^^^^^^^^^^^^ not found in this scope
   |
help: consider importing one of these unit variants
   |
1  + use crate::software::scaling::SwsFlags::SWS_BICUBLIN;
   |
1  + use ffmpeg_sys_the_third::SwsFlags::SWS_BICUBLIN;
   |

error[E0425]: cannot find value `SWS_GAUSS` in this scope
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:13:38
   |
13 |         const GAUSS                = SWS_GAUSS;
   |                                      ^^^^^^^^^ not found in this scope
   |
help: consider importing one of these unit variants
   |
1  + use crate::software::scaling::SwsFlags::SWS_GAUSS;
   |
1  + use ffmpeg_sys_the_third::SwsFlags::SWS_GAUSS;
   |

error[E0425]: cannot find value `SWS_SINC` in this scope
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:14:38
   |
14 |         const SINC                 = SWS_SINC;
   |                                      ^^^^^^^^ not found in this scope
   |
help: consider importing one of these unit variants
   |
1  + use crate::software::scaling::SwsFlags::SWS_SINC;
   |
1  + use ffmpeg_sys_the_third::SwsFlags::SWS_SINC;
   |

error[E0425]: cannot find value `SWS_LANCZOS` in this scope
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:15:38
   |
15 |         const LANCZOS              = SWS_LANCZOS;
   |                                      ^^^^^^^^^^^ not found in this scope
   |
help: consider importing one of these unit variants
   |
1  + use crate::software::scaling::SwsFlags::SWS_LANCZOS;
   |
1  + use ffmpeg_sys_the_third::SwsFlags::SWS_LANCZOS;
   |

error[E0425]: cannot find value `SWS_SPLINE` in this scope
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:16:38
   |
16 |         const SPLINE               = SWS_SPLINE;
   |                                      ^^^^^^^^^^ not found in this scope
   |
help: consider importing one of these unit variants
   |
1  + use crate::software::scaling::SwsFlags::SWS_SPLINE;
   |
1  + use ffmpeg_sys_the_third::SwsFlags::SWS_SPLINE;
   |

error[E0425]: cannot find value `SWS_PRINT_INFO` in this scope
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:20:38
   |
20 |         const PRINT_INFO           = SWS_PRINT_INFO;
   |                                      ^^^^^^^^^^^^^^ not found in this scope
   |
help: consider importing one of these unit variants
   |
1  + use crate::software::scaling::SwsFlags::SWS_PRINT_INFO;
   |
1  + use ffmpeg_sys_the_third::SwsFlags::SWS_PRINT_INFO;
   |

error[E0425]: cannot find value `SWS_FULL_CHR_H_INT` in this scope
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:21:38
   |
21 |         const FULL_CHR_H_INT       = SWS_FULL_CHR_H_INT;
   |                                      ^^^^^^^^^^^^^^^^^^ not found in this scope
   |
help: consider importing one of these unit variants
   |
1  + use crate::software::scaling::SwsFlags::SWS_FULL_CHR_H_INT;
   |
1  + use ffmpeg_sys_the_third::SwsFlags::SWS_FULL_CHR_H_INT;
   |

error[E0425]: cannot find value `SWS_FULL_CHR_H_INP` in this scope
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:22:38
   |
22 |         const FULL_CHR_H_INP       = SWS_FULL_CHR_H_INP;
   |                                      ^^^^^^^^^^^^^^^^^^ not found in this scope
   |
help: consider importing one of these unit variants
   |
1  + use crate::software::scaling::SwsFlags::SWS_FULL_CHR_H_INP;
   |
1  + use ffmpeg_sys_the_third::SwsFlags::SWS_FULL_CHR_H_INP;
   |

error[E0425]: cannot find value `SWS_DIRECT_BGR` in this scope
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:23:38
   |
23 |         const DIRECT_BGR           = SWS_DIRECT_BGR;
   |                                      ^^^^^^^^^^^^^^ not found in this scope
   |
help: consider importing one of these unit variants
   |
1  + use crate::software::scaling::SwsFlags::SWS_DIRECT_BGR;
   |
1  + use ffmpeg_sys_the_third::SwsFlags::SWS_DIRECT_BGR;
   |

error[E0425]: cannot find value `SWS_ACCURATE_RND` in this scope
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:24:38
   |
24 |         const ACCURATE_RND         = SWS_ACCURATE_RND;
   |                                      ^^^^^^^^^^^^^^^^ not found in this scope
   |
help: consider importing one of these unit variants
   |
1  + use crate::software::scaling::SwsFlags::SWS_ACCURATE_RND;
   |
1  + use ffmpeg_sys_the_third::SwsFlags::SWS_ACCURATE_RND;
   |

error[E0425]: cannot find value `SWS_BITEXACT` in this scope
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:25:38
   |
25 |         const BITEXACT             = SWS_BITEXACT;
   |                                      ^^^^^^^^^^^^ not found in this scope
   |
help: consider importing one of these unit variants
   |
1  + use crate::software::scaling::SwsFlags::SWS_BITEXACT;
   |
1  + use ffmpeg_sys_the_third::SwsFlags::SWS_BITEXACT;
   |

error[E0425]: cannot find value `SWS_ERROR_DIFFUSION` in this scope
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ffmpeg-the-third-2.0.1+ffmpeg-7.0/src/software/scaling/flag.rs:26:38
   |
26 |         const ERROR_DIFFUSION      = SWS_ERROR_DIFFUSION;
   |                                      ^^^^^^^^^^^^^^^^^^^ not found in this scope
   |
help: consider importing one of these unit variants
   |
1  + use crate::software::scaling::SwsFlags::SWS_ERROR_DIFFUSION;
   |
1  + use ffmpeg_sys_the_third::SwsFlags::SWS_ERROR_DIFFUSION;
   |

For more information about this error, try `rustc --explain E0425`.
error: could not compile `ffmpeg-the-third` (lib) due to 18 previous errors
warning: build failed, waiting for other jobs to finish...
/sbin/ldconfig.real: /lib/libavdevice.so.61 is not a symbolic link

/sbin/ldconfig.real: /lib/libavformat.so.61 is not a symbolic link

/sbin/ldconfig.real: /lib/libavutil.so.59 is not a symbolic link

/sbin/ldconfig.real: /lib/libpostproc.so.58 is not a symbolic link

/sbin/ldconfig.real: /lib/libavcodec.so.61 is not a symbolic link

/sbin/ldconfig.real: /lib/libswresample.so.5 is not a symbolic link

/sbin/ldconfig.real: /lib/libswscale.so.8 is not a symbolic link

/sbin/ldconfig.real: /lib/libavfilter.so.10 is not a symbolic link
```

**Comment by master-of-zen** [comment]
Hmmm, from the looks of it, there is something with ffmpeg messed up.
Are you sure you're using latest version?

**Comment by LigH-de** [comment]
Ubuntu is not a reason, it fails in MSYS2/MinGW too. 

All the failing functions are related to software scaling. I believe ffmpeg just increased its minor version to 7.2.x, maybe there are related changes.

**Comment by arshlearn** [comment]
> Hmmm, from the looks of it, there is something with ffmpeg messed up. Are you sure you're using latest version?

yes i am using the latest shared build from @Uranite

**Comment by arshlearn** [comment]
nvm guys, i fixed it by using ffmpeg-7.1 instead of master

**Comment by LigH-de** [comment]
Well, nice for you, @arshlearn ... but for all the others who try to apply master/HEAD (e.g. users of M-AB-S), it is not done yet...
